python/tests/test_events.py

```diff
@@ -298,17 +309,260 @@ def get_early_x(t):
 
         return x.flatten()
 
-    def sx_pected(t, parameters):
-        # get sx, w.r.t. parameters, via finite differences
+    def sx_expected(t, parameters):
+        """get sx, w.r.t. parameters, via finite differences"""
+        sx = []
+        eps = 1e-6
+
+        for ip in parameters:
+            perturbed_params = deepcopy(parameters)
+            perturbed_params[ip] += eps
+            sx_p = x_expected(t, **perturbed_params)
+            perturbed_params[ip] -= 2 * eps
+            sx_m = x_expected(t, **perturbed_params)
+            sx.append((sx_p - sx_m) / (2 * eps))
+
+        return np.array(sx)
+
+    return (
+        initial_assignments,
+        parameters,
+        rate_rules,
+        species,
+        events,
+        timepoints,
+        x_expected,
+        sx_expected,
+    )
+
+
+def model_definition_piecewise_plus_event_simple_case():
+    """Test model for boolean operations in a piecewise condition.
+
+    ODEs
+    ----
+    d/dt x_1:
+        - { 1,    (alpha <= t and t < beta)
+        - { 0,    otherwise
+    """
+    # Model components
+    species = ["x_1"]
+    initial_assignments = {"x_1": "x_1_0"}
+    rate_rules = {"x_1": "piecewise(1, (alpha < time && time < beta), 0)"}
+    parameters = {
+        "alpha": 2,
+        "beta": 3,
+        "gamma": 4.5,
+        "x_1_0": 1,
+    }
+    timepoints = np.linspace(0.0, 5.0, 100)  # np.array((0.0, 4.0,))
+    events = {
+        "event_1": {"trigger": "time > alpha", "target": "x_1", "assignment": "gamma"},
+        "event_2": {
+            "trigger": "time > beta",
+            "target": "x_1",
+            "assignment": "x_1 + 2.5",
+        },
+    }
+
+    # Analytical solution
+    def x_expected(t, x_1_0, alpha, beta, gamma):
+        t_event_1 = alpha
+        t_event_2 = beta
+
+        if t < t_event_1:
+            x = x_1_0
+        elif t < t_event_2:
+            x = gamma + t - t_event_1
+        else:
+            x = gamma + t_event_2 - t_event_1 + 2.5
+
+        return np.array((x,))
+
+    def sx_expected(t, parameters):
+        """get sx, w.r.t. parameters, via finite differences"""
+        sx = []
+        eps = 1e-6
+
+        for ip in parameters:
+            perturbed_params = deepcopy(parameters)
+            perturbed_params[ip] += eps
+            sx_p = np.array(x_expected(t, **perturbed_params))
+            perturbed_params[ip] -= 2 * eps
+            sx_m = np.array(x_expected(t, **perturbed_params))
+            sx.append((sx_p - sx_m) / (2 * eps))
+
+        return np.array(sx)
+
+    return (
+        initial_assignments,
+        parameters,
+        rate_rules,
+        species,
+        events,
+        timepoints,
+        x_expected,
+        sx_expected,
+    )
+
+
+def model_definition_piecewise_plus_event_semi_complicated():
+    """Test model for boolean operations in a piecewise condition, discrete
+    events and a non-vanishing quadrature for the adjoint state.
+    """
```

Comment:
Currently fails finite differences check. 

Passes with 
```diff
diff --git a/python/tests/util.py b/python/tests/util.py
index d5907da29..e85160a98 100644
--- a/python/tests/util.py
+++ b/python/tests/util.py
@@ -154,4 +154,4 @@ def check_trajectories_with_forward_sensitivities(
     solver.setRelativeToleranceFSA(1e-13)
     rdata = runAmiciSimulation(amici_model, solver=solver)
     _check_close(rdata["x"], result_expected_x, field="x", rtol=1e-10, atol=1e-12)
-    _check_close(rdata["sx"], result_expected_sx, field="sx", rtol=1e-10, atol=1e-9)
+    _check_close(rdata["sx"], result_expected_sx, field="sx", rtol=1e-07, atol=1e-9)
```

Seems a bit loose, but then again, we are comparing to FD...